modules/helper/main.tf

```diff
@@ -69,15 +87,30 @@ locals {
     ]
   ]))
 
+  keys_additive_conditional = distinct(flatten([
+    for alias in local.aliased_entities
+    : [
+      for binding in var.conditional_bindings
+      : [
+        for member in binding.members
+        : "${alias}--${binding.role}--${binding.title}--${member}"
+      ]
+    ]
+  ]))
+
+  all_keys_authoritative = concat(local.keys_authoritative, local.keys_authoritative_conditional)
+
+  all_keys_additive = concat(local.keys_additive, local.keys_additive_conditional)
+
   bindings_authoritative = (
     local.authoritative
-    ? zipmap(local.keys_authoritative, local.bindings_by_role)
+    ? zipmap(local.all_keys_authoritative, local.bindings_by_role)
```

Comment:
for `bindings_authoritative` we are zipping `all_keys_authoritative` which includes keys from both regular and conditionals but `bindings_by_role` only has regular bindings. Shouldn't we be zipping with `all_bindings`?

Reply:
Good catch. It should be `all_bindings`. Also about test cases, I am going to add a couple cases for project and bucket maybe or you think it would make sense to add a separate test case with just conditional bindings for all the supported resources?

Reply:
IMO it should be sufficient to have a few test cases. We don't need to be exhaustive.